tests/CoreStubsTest.php

```diff
@@ -131,5 +133,35 @@ function foo(string $foo): string
                 '$a===' => 'non-empty-string',
             ],
         ];
+        yield 'json_encode returns a non-empty-string with JSON_THROW_ON_ERROR' => [
+            'code' => '<?php
+                $a = json_encode([], JSON_THROW_ON_ERROR | JSON_HEX_TAG);
+                $b = json_encode([], JSON_THROW_ON_ERROR | JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_QUOT | JSON_HEX_AMP | JSON_UNESCAPED_UNICODE);
+                $c = json_encode([], JSON_THROW_ON_ERROR | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
+                $d = json_encode([], JSON_THROW_ON_ERROR | JSON_PRESERVE_ZERO_FRACTION);
+            ',
+            'assertions' => [
+                '$a===' => 'non-empty-string',
+                '$b===' => 'non-empty-string',
+                '$c===' => 'non-empty-string',
+                '$d===' => 'non-empty-string',
```

Comment:
Do we have any tests for `non-empty-string|false` variant?

Reply:
Nice catch. Will add some.
Actually there is also `JSON_PARTIAL_OUTPUT_ON_ERROR` that narrows the type to `non-empty-string`, but adding support for it requires adding 10 condition branches for `int<512,1023>`, `int<1536,2047>` and so on 😅, so I'm not really sure is it worth it.